WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewModel.swift

```diff
@@ -239,19 +231,57 @@ extension ProductFormViewModel {
 //
 extension ProductFormViewModel {
     func updateProductRemotely(onCompletion: @escaping (Result<EditableProductModel, ProductUpdateError>) -> Void) {
-        let updateProductAction = ProductAction.updateProduct(product: product.product) { [weak self] result in
-            switch result {
-            case .failure(let error):
-                ServiceLocator.analytics.track(.productDetailUpdateError, withError: error)
-                onCompletion(.failure(error))
-            case .success(let product):
-                ServiceLocator.analytics.track(.productDetailUpdateSuccess)
-                let model = EditableProductModel(product: product)
-                self?.resetProduct(model)
-                onCompletion(.success(model))
+        let remoteActionUseCase = ProductFormRemoteActionUseCase()
+        switch formType {
+        case .add:
+            remoteActionUseCase.addProduct(product: product, password: password) { [weak self] productResult, passwordResult in
```

Comment:
Just passing one result actually sounds cleaner keeping the two results internal, thanks for the suggestion! (like if one day Woo API supports updating the password in the product update endpoint) updated in https://github.com/woocommerce/woocommerce-ios/pull/2785/commits/ec900ac19483a08c697256300524b4cf15b86bcf (I created a struct `ResultData` as a wrapper for the product and password, since a tuple can't be compared in unit tests)